src/with-authentication-required.tsx

```diff
@@ -60,6 +69,11 @@ export interface WithAuthenticationRequiredOptions {
    * This will be merged with the `returnTo` option used by the `onRedirectCallback` handler.
    */
   loginOptions?: RedirectLoginOptions;
+  /**
+   * Check the user object for JWT claims and return a boolean indicating
+   * whether or not they are authorized to view the component.
+   */
+  claimCheck?: (claims: JWTNamespaces | User) => boolean;
```

Comment:
```suggestion
  claimCheck?: (claims: User) => boolean;
```
And please could you delete the `JWTClaim` and `JWTNamespaces` types